be/src/vec/data_types/data_type_struct.cpp

```diff
@@ -355,14 +355,7 @@ std::optional<size_t> DataTypeStruct::try_get_position_by_name(const String& nam
 }
 
 String DataTypeStruct::get_name_by_position(size_t i) const {
-    if (i == 0 || i > names.size()) {
```

Comment:
Why removing this check?

Reply:
This function is wrong, and there's no check like `Block::get_by_position`